spark3/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -132,10 +134,17 @@ protected TableIdentifier buildIdentifier(Identifier identifier) {
     return TableIdentifier.of(Namespace.of(identifier.namespace()), identifier.name());
   }
 
+  private String[] currentNamespace() {
+    return SparkSession.active().sessionState().catalogManager().currentNamespace();
```

Comment:
If this uses a session, then it should be identified when the catalog is created and stored as an instance field. Catalogs are specific to a SQL session, so they should not use the active session except for in initialization. After that, the same session should always be used.

Reply:
fixed. Just for my learning why is it ok to get hadoop config from active session but not the current namespace? I have been following the commonly used `SparkSession.active().sessionState().newHadoopConf()` pattern